File: convert_pdf.py
import fitz # PyMuPDF
import os

pdf_path = r"c:\billing-system\sample_invoice.pdf"
output_path = r"C:\Users\Admin\.gemini\antigravity-ide\brain\8b5523b8-dcfa-461d-a086-0b85750febd2\sample_invoice.png"

try:
    print(f"Loading PDF: {pdf_path}")
    doc = fitz.open(pdf_path)
    page = doc.load_page(0) # first page
    
    # render page to a pixmap (image) at high resolution (150 dpi)
    pix = page.get_pixmap(dpi=150)
    print(f"Saving PNG image to: {output_path}")
    pix.save(output_path)
    
    print("\n======================================================")
    print("SUCCESS: Real PDF converted to PNG successfully via PyMuPDF!")
    print(f"Image saved to: {output_path}")
    print("======================================================\n")
except Exception as e:
    print("Conversion failed:", e)

File: backend/convert_pdf.js
const pdf2img = require('pdf-img-convert');
const fs = require('fs');
const path = require('path');

async function convert() {
    try {
        const pdfPath = path.join(__dirname, "..", "sample_invoice.pdf");
        console.log(`Loading PDF from: ${pdfPath}`);
        
        // Convert to images
        const outputImages = await pdf2img.convert(pdfPath, {
            width: 1000 // high-resolution width
        });
        
        // Save the first page as PNG
        const targetPath = "C:\\Users\\Admin\\.gemini\\antigravity-ide\\brain\\8b5523b8-dcfa-461d-a086-0b85750febd2\\sample_invoice.png";
        fs.writeFileSync(targetPath, outputImages[0]);
        console.log(`\n======================================================`);
        console.log(`SUCCESS: Real PDF converted to PNG image!`);
        console.log(`File saved to: ${targetPath}`);
        console.log(`======================================================\n`);
    } catch (error) {
        console.error("Conversion failed:", error);
    }
}

convert();
